sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,271 @@
+        3. Creates tasks for partitions that are newly claimed by this EventProcessor
+        4. Keeps tasks running for partitions that haven't changed ownership
+        5. Each task repeatedly calls EvenHubConsumer.receive() to retrieve events and
+        call user defined partition processor
+
+        :return: None
+
+        """
+        log.info("EventProcessor %r is being started", self._id)
+        ownership_manager = OwnershipManager(self._eventhub_client, self._consumer_group_name, self._id,
+                                             self._partition_manager, self._ownership_timeout)
+        if not self._running:
+            self._running = True
+            while self._running:
+                try:
+                    claimed_ownership_list = await ownership_manager.claim_ownership()
+                except Exception as err:  # pylint:disable=broad-except
+                    log.warning("An exception (%r) occurred during balancing and claiming ownership for eventhub %r "
+                                "consumer group %r. Retrying after %r seconds",
+                                err, self._eventhub_name, self._consumer_group_name, self._polling_interval)
+                    await asyncio.sleep(self._polling_interval)
+                    continue
+
+                to_cancel_list = self._tasks.keys()
+                if claimed_ownership_list:
+                    claimed_partition_ids = [x["partition_id"] for x in claimed_ownership_list]
+                    to_cancel_list = self._tasks.keys() - claimed_partition_ids
+                    self._create_tasks_for_claimed_ownership(claimed_ownership_list)
+                else:
+                    log.info("EventProcessor %r hasn't claimed an ownership. It keeps claiming.", self._id)
+                if to_cancel_list:
+                    self._cancel_tasks_for_partitions(to_cancel_list)
+                    log.info("EventProcesor %r has cancelled partitions %r", self._id, to_cancel_list)
+                await asyncio.sleep(self._polling_interval)
+
+    async def stop(self):
+        """Stop claiming ownership and all the partition consumers owned by this EventProcessor
+
+        This method stops claiming ownership of owned partitions and cancels tasks that are running
+        EventHubConsumer.receive() for the partitions owned by this EventProcessor.
+
+        :return: None
+
+        """
+        self._running = False
+        for _ in range(len(self._tasks)):
+            _, task = self._tasks.popitem()
+            task.cancel()
+        log.info("EventProcessor %r has been cancelled", self._id)
+        await asyncio.sleep(2)  # give some time to finish after cancelled.
+
+    def _cancel_tasks_for_partitions(self, to_cancel_partitions):
+        for partition_id in to_cancel_partitions:
+            if partition_id in self._tasks:
+                task = self._tasks.pop(partition_id)
+                task.cancel()
```

Comment:
I see in line 174: `await asyncio.sleep(2)  # give some time to finish after cancelled.`, do we need to do the same `asyncio.sleep` here?